File: Objective-C/AlertController_ObjC/Utility/Utility.m
//
//  Utility.m
//  AlertController_ObjC
//
//  Created by Hardik Trivedi on 06/11/19.
//  Copyright © 2019 9Brainz. All rights reserved.
//

#import "Utility.h"

@implementation Utility

//
//    // INPUT Parameter
//
//    // vc       : ViewController (in which you want to display Alert)
//    // title    : Title text of AlertController
//    // message  : Message text of AlertController
//    // button   : Multiple buttons for AlertController
//

+ (void)
shwoAlertControllerInViewController: (UIViewController *)vc withTitle: (nullable NSString *)title andMessage: (nullable NSString *)message withButtons: (NSArray *)button withCompletion: (completion)completionBlock
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
    
    for (int index = 0; index < button.count; index++) {
        
        UIAlertAction *action = [UIAlertAction actionWithTitle:button[index] style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
            
            if (completionBlock != nil) {

                completionBlock(index);
            }
        }];
        
        [alertController addAction:action];
    }
    
    [vc presentViewController:alertController animated:YES completion:nil];
}

@end

File: Swift/AlertController_Swift/Utility/Utility.swift
//
//  Utility.swift
//  AlertController_Swift
//
//  Created by Hardik Trivedi on 06/11/19.
//  Copyright © 2019 9Brainz. All rights reserved.
//

import UIKit

class Utility: NSObject
{
    
    // INPUT Parameter
    
    // vc       : ViewController (in which you want to display Alert)
    // title    : Title text of AlertController
    // message  : Message text of AlertController
    // button   : Multiple buttons for AlertController
    
    static func shwoAlertControllerInViewController(vc: UIViewController, withTitle title: String?, andMessage message: String?, withButtons button: [String], completion:((_ index: Int) -> Void)!) -> Void
    {
        let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)

        for index in 0 ..< button.count  {

            let action = UIAlertAction(title: button[index], style: .default, handler: { (alert: UIAlertAction!) in

                if completion != nil {

                    completion(index)
                }
            })

            alertController.addAction(action)
        }

        vc.present(alertController, animated: true, completion: nil)
    }

}
